lib/lens/sender.rb

```diff
@@ -3,66 +3,40 @@
 
 module Lens
   class Sender
-    NOTICES_URI = 'api/v1/events'
-    HTTP_ERRORS = [Timeout::Error,
-                   Errno::EINVAL,
-                   Errno::ECONNRESET,
-                   EOFError,
-                   Net::HTTPBadResponse,
-                   Net::HTTPHeaderSyntaxError,
-                   Net::ProtocolError,
-                   Errno::ECONNREFUSED].freeze
+    NOTICES_URI = 'api/v1/events'.freeze
 
-    def initialize(options = {})
-      [ :app_key,
-        :protocol,
-        :host,
-        :port
-      ].each do |option|
-        instance_variable_set("@#{option}", options.send(option))
-      end
-    end
-
-    def send_to_lens(data)
-      send_request(url.path, data)
-    end
+    attr_reader :app_key, :protocol, :host, :port
 
-    attr_reader :app_key,
-      :protocol,
-      :host,
-      :port
+    def initialize(options = nil)
+      raise ArgumentError unless options.is_a? Configuration
 
-  private
-
-    def http_connection
-      setup_http_connection
+      @app_key = options.app_key
+      @protocol = options.protocol
+      @protocol = options.protocol
+      @host = options.host
+      @port = options.port
     end
 
-    def setup_http_connection
-      Net::HTTP.new(url.host, url.port)
-    rescue => e
-      log(:error, "[Lens::Sender#setup_http_connection] Failure initializing the HTTP connection.\nError: #{e.class} - #{e.message}\nBacktrace:\n#{e.backtrace.join("\n\t")}")
-      raise e
+    def send_to_lens(data)
+      send_request(data, url.path)
     end
 
-    def send_request(path, data, headers = {})
-      http_connection.post(path, data, http_headers(headers))
-    rescue *HTTP_ERRORS => e
-      raise e
-      nil
+    private
+
+    def send_request(data, path)
+      http_connection.post(path, data, http_headers)
     end
 
-    def http_headers(headers=nil)
-      {}.tap do |hash|
-        hash.merge!(HEADERS)
-        hash.merge!({'X-Auth-Token' => app_key})
-        hash.merge!({'Content-Type' =>'application/json'})
-        hash.merge!(headers) if headers
-      end
+    def http_headers
+      HEADERS.merge({ 'X-Auth-Token' => app_key })
```

Comment:
Redundant curly braces around a hash parameter.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.